flow/ThreadPrimitives.h

```diff
@@ -44,7 +45,11 @@
 #include <drd.h>
 #endif
 
-class ThreadSpinLock {
+// TODO: We should make this dependent on the CPU. Maybe cmake
+// can set this variable properly?
+constexpr size_t CACHE_LINE_SIZE = 64;
```

Comment:
`std::hardware_destructive_interference_size`

(and maybe delete CACHE_LINE_SIZE, since it's only used twice).

Documentation:
https://en.cppreference.com/w/cpp/thread/hardware_destructive_interference_size

Reply:
This doesn't seem to be available on macOS:

```
/Users/mpilman/Projects/foundationdb/flow/ThreadPrimitives.h:49:20: error: no member named 'hardware_destructive_interference_size' in namespace 'std'
class alignas(std::hardware_destructive_interference_size) ThreadSpinLock {
```

It does not seem to be in [libc++](https://libcxx.llvm.org/cxx1z_status.html) yet :(

It would be possible to use cmake to check whether this exists in the current tool chain, but as 64 seems to be pretty standard these days, I would prefer to keep the TODO there and revisit when this is more broadly supported.

Thoughts @sears?